scripts/protect-branches.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+
+import argparse
+import json
+import os
+import re
+import requests
+import sys
+import urllib
+
+
+# API token received from github. For the "scopes", I only selected the entire "repo" section.
+# https://github.com/settings/tokens
+TOKEN=""
+
+
+# the Accept header tells github to let us use their experimental branch protection API
+headers = {
+    'Authorization': 'token {}'.format(TOKEN),
+    'Accept': 'application/vnd.github.loki-preview+json',
+}
+
+
+# the body to send when setting protection on a branch
+protection_body = json.dumps({'restrictions': None, 'required_status_checks': None})
+
+
+name_re = re.compile('^master|.*-dev|.*-release$')
+
+
+PULP_REPOS_URL = "https://api.github.com/repos/pulp/"
+
+
+repo_names = (
+    'crane',
+    'devel',
+    'nectar',
+    'pulp',
+    'pulp_docker',
+    'pulp_ostree',
+    'pulp_puppet',
+    'pulp_python',
+    'pulp_rpm',
+)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Sets branches on GitHub to be protected. '
+                                     'Operates on master, *-release, and *-dev branches.')
+    parser.add_argument('-l', '--list', action='store_true',
+                        help='List branches that should be protected without making any changes.')
```

Comment:
I tried this option and the explanation didn't match what it's behavior does. I'll suggest this replacement:

`List branches that should be protected but are not. This command makes no changes.`